src/controller/CHIPDeviceController.cpp

```diff
@@ -3572,17 +3578,34 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
             return;
         }
 
-        CHIP_ERROR err = ValidateJFAdminNOC(params.GetJFAdminNOC().Value());
+        P256PublicKeySpan jfAdminICACPKSpan;
+        CHIP_ERROR err = CHIP_NO_ERROR;
+
+        /* check the peer Admin NOC */
+        err = ValidateJFAdminNOC(params.GetJFAdminNOC().Value());
```

Comment:
These changes seem to conflict with https://github.com/project-chip/connectedhomeip/pull/39250... which one is happening?

Reply:
I think @robertfarnum missed the fact that verification of peer Admin NOC is already in place and it can be reused inside his refactoring work. Left also a comment for his PR: https://github.com/project-chip/connectedhomeip/pull/39250#discussion_r2120671620